src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        const StateBfAdd = { ...(newStates[newStates.length - 1] || state) };
```

Comment:
Italo, a variable should start with a lowercase letter, we use uppercase for classes, types and such, but not variables.
Also, `stateBfAdd` isn't very intuitive, you could call it something like `previousState`.
The same advice works for other declarations